lib/instrumentation/@elastic/elasticsearch.js

```diff
@@ -37,20 +38,29 @@ function queryParser(params) {
   let collection
 
   const suffix = actions[params.method]
-  path.forEach((segment, idx) => {
-    const prev = idx - 1
-    if (segment === '_search') {
-      collection = path?.[prev] || defaultCollection
-      operation = `${segment}`
-    } else if (segment[0] === '_') {
-      collection = path?.[prev] || defaultCollection
-      operation = `${segment}.${suffix}`
+  try {
+    path.forEach((segment, idx) => {
```

Comment:
this code is still problematic where a user tries to create an index that starts with a `_`.

If I try to create `_foo`. the span that gets created has a name of `Datastore/statement/ElasticSearch/any/foo.update`

Reply:
It's even more confusing if you try to create an index for something like `_search`

Reply:
I'd think the only way we have of guaranteeing a valid ES command is to compare against a list of known valid commands, similarly to the Go agent. Then `_foo` would fall through to the defaults, if that's what we want.

Reply:
On thinking it through, it makes more sense to capture the path for an index creation that doesn't work. Updated.